mubench.pipeline/tasks/configurations/configurations.py

```diff
@@ -0,0 +1,196 @@
+from typing import List
+
+from data.detectors import find_detector
+from data.experiments import ProvidedPatternsExperiment, TopFindingsExperiment, BenchmarkExperiment
+from tasks.configurations.mubench_paths import MubenchPaths
+from tasks.implementations import stats
+from tasks.implementations.checkout import CheckoutTask
+from tasks.implementations.collect_misuses import CollectMisusesTask
+from tasks.implementations.collect_projects import CollectProjectsTask
+from tasks.implementations.collect_versions import CollectVersionsTask
+from tasks.implementations.compile import CompileTask
+from tasks.implementations.compile_patterns import CompilePatternsTask
+from tasks.implementations.dataset_check import DatasetCheckTask
+from tasks.implementations.detect import DetectTask
+from tasks.implementations.info import ProjectInfoTask, VersionInfoTask, MisuseInfoTask
+from tasks.implementations.publish_findings import PublishFindingsTask
+from tasks.implementations.publish_metadata import PublishMetadataTask
+from tasks.task_runner import TaskRunner
+from utils.dataset_util import get_available_datasets
+
+
+class TaskConfiguration(list):
+    def __init__(self, config):
+        super().__init__()
+        self.extend(self._tasks(config))
+
+    @staticmethod
+    def mode() -> str:
+        raise NotImplementedError
+
+    def _tasks(self, config) -> List:
+        raise NotImplementedError
+
+
+def get_task_configuration(config) -> TaskConfiguration:
```

Comment:
We don't really need to pass the `TaskConfiguration` to the outer world, right? All that the caller cares about is a list of tasks. How we realize the mapping from modes to tasks is an implementation detail. Therefore, I would call this function `create_tasks` and return `requested_configuration.tasks()` from it. This would also get us rid of the `self.extend()` call in the constructor, a potential cause of race conditions.